tests/rptest/tests/cluster_self_config_test.py

```diff
@@ -0,0 +1,103 @@
+        si_settings = SISettings(
+            ctx,
+            #Force self configuration through setting cloud_storage_url_style to None.
+            cloud_storage_url_style=None,
+            cloud_storage_enable_remote_read=ctx.
+            injected_args["cloud_storage_enable_remote_read"],
+            cloud_storage_enable_remote_write=ctx.
+            injected_args["cloud_storage_enable_remote_write"])
+
+        super(ClusterSelfConfigTest, self).__init__(ctx,
+                                                    si_settings=si_settings)
+
+        self.log_searcher = LogSearchLocal(ctx, [], self.redpanda.logger,
+                                           self.redpanda.STDOUT_STDERR_CAPTURE)
+        self.admin = Admin(self.redpanda)
+
+    @cluster(num_nodes=1)
+    @matrix(cloud_storage_enable_remote_read=[True, False],
+            cloud_storage_enable_remote_write=[True, False])
+    def test_s3_self_config(self, cloud_storage_enable_remote_read,
+                            cloud_storage_enable_remote_write):
+        """
+        Verify that cloud_storage_url_style self configuration occurs for the s3_client
+        when it is not specified. There aren't any endpoints for testing this, so
+        it will be manually checked for from the logs.
+        """
+
+        config = self.admin.get_cluster_config()
+
+        # Even after self-configuring, the cloud_storage_url_style setting will
+        # still be left unset at the cluster config level.
+        assert config['cloud_storage_url_style'] == None
+
+        def self_config_start_in_logs(node):
+            client_self_configuration_start_string = 'Client requires self configuration step'
+            for log in self.log_searcher._capture_log(
+                    node, client_self_configuration_start_string):
+                m = re.search(client_self_configuration_start_string,
+                              log.strip())
+                if m:
+                    return True
+            return False
+
+        def self_config_result_from_logs(node):
+            client_self_configuration_complete_string = 'Client self configuration completed with result'
+            for log in self.log_searcher._capture_log(
+                    node, client_self_configuration_complete_string):
+                m = re.search(
+                    client_self_configuration_complete_string + r' (\{.*\})',
+                    log.strip())
+                if m:
+                    return (True, m.group(1))
+            return (False, None)
+
+        def self_config_default_in_logs(node):
+            client_self_configuration_default_string = r'Can\'t self-configure S3 Client'
+            for log in self.log_searcher._capture_log(
+                    node, client_self_configuration_default_string):
+                m = re.search(client_self_configuration_default_string,
+                              log.strip())
+                if m:
+                    return True
```

Comment:
nit: same question as above re. regex usage. Also escaping characters can be avoided with

```
client_self_configuration_default_string = """Can't self-configure S3 Client"""
```

Reply:
From my testing, using the raw string works best with `_capture_log()`.

Reply:
Updated `redpanda` side message from `Can't ...` to `Could not...` for ease of parsing.